webui/react/src/components/Searches/Searches.tsx

```diff
@@ -309,12 +304,20 @@ const Searches: React.FC<Props> = ({ project }) => {
     [resetPagination, sortString, updateSettings],
   );
 
-  useEffect(() => {
-    if (!isLoadingSettings && settings.sortString) {
-      setSorts(parseSortString(settings.sortString));
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isLoadingSettings]);
+  useLayoutEffect(() => {
+    let cleanup: () => void;
+    // eslint-disable-next-line prefer-const
+    cleanup = eagerSubscribe(projectSettingsObs, (ps, prevPs) => {
```

Comment:
q: what's the reason for declaring the variable separately from initializing it?

Reply:
ran into an issue where `cleanup` wasn't defined in scope when the module was parsed. when we used this pattern for filterformstate, cleanup was defined and set at runtime, so i think it's just an odd issue. I'm going to take another look and see if it still makes sense.